src/isolineFeature.js

```diff
@@ -0,0 +1,891 @@
+          totalDistance = 0,
+          dist, count, localSpacing, next, lineDistance, i, i2, f, g, pos,
+          mapCoor;
+      if (spacing <= 0 || isNaN(spacing)) {
+        return;
+      }
+      maxSpacing = Math.max(spacing, maxSpacing);
+      /* make offset in the range of [0, 1) with the default at 0.5 */
+      offset = (offset + 0.5) - Math.floor(offset + 0.5);
+      dist = dispCoor.map(function (pt1, coorIdx) {
+        if (!line.closed && coorIdx + 1 === dispCoor.length) {
+          return 0;
+        }
+        var val = Math.sqrt(util.distance2dSquared(pt1, dispCoor[coorIdx + 1 < dispCoor.length ? coorIdx + 1 : 0]));
+        totalDistance += val;
+        return val;
+      });
+      count = Math.floor(totalDistance / spacing);
+      if (!count) {
+        return;
+      }
+      /* If we have any labels, compute map coordinates of the line and use
+       * those for interpolating label positions */
+      mapCoor = transform.transformCoordinates(gcs, mapgcs, line);
+      localSpacing = totalDistance / count;
+      next = localSpacing * offset;
+      lineDistance = 0;
+      for (i = 0; i < dispCoor.length; i += 1) {
+        while (lineDistance + dist[i] >= next) {
+          i2 = i + 1 === dispCoor.length ? 0 : i + 1;
+          f = (next - lineDistance) / dist[i];
+          g = 1 - f;
+          next += localSpacing;
+          if (labelViewport > 0) {
+            pos = {
+              x: dispCoor[i].x * g + dispCoor[i2].x * f,
+              y: dispCoor[i].y * g + dispCoor[i2].y * f
+            };
+            if (pos.x < -labelViewport || pos.x > mapSize.width + labelViewport ||
+                pos.y < -labelViewport || pos.y > mapSize.height + labelViewport) {
+              continue;
+            }
+          }
+          labelData.push({
+            x: mapCoor[i].x * g + mapCoor[i2].x * f,
+            y: mapCoor[i].y * g + mapCoor[i2].y * f,
+            z: mapCoor[i].z * g + mapCoor[i2].z * f,
+            line: line,
+            rotation: Math.atan2(dispCoor[i].y - dispCoor[i2].y, dispCoor[i].x - dispCoor[i2].x) - mapRotation
+          });
+        }
+        lineDistance += dist[i];
+      }
+    });
+    m_labelFeature.gcs(mapgcs);
+    m_labelFeature.data(labelData);
+    m_labelFeature.style('renderThreshold', maxSpacing * 2);
+    m_lastLabelPositions = {
+      zoom: map.zoom(),
+      center: map.center(),
+      rotation: mapRotation,
+      size: mapSize,
+      labelViewport: labelViewport,
+      maxSpacing: maxSpacing,
+      labelAutoUpdate: isoline.get('labelAutoUpdate')(m_isolines.mesh)
+    };
+    return m_this;
+  };
+
+  /**
+   * Get the last map position that was used for generating labels.
+   *
+   * @returns {object} An object with the map `zoom` and `center` and the
+   *    `labelViewport` used in generating labels.  The object may have no
+   *    properties if there are no labels.
+   */
+  this.lastLabelPositions = function () {
```

Comment:
Same here, not entirely sure if we need this in the public API

Reply:
Yes.  As with `labelPositions`, this information can be used to decide when to regenerate labels if the default is inappropriate.